lib/productions/type.js

```diff
@@ -194,6 +194,25 @@ export class Type extends Base {
   *validate(defs) {
     yield* this.extAttrs.validate(defs);
 
+    if (this.idlType === "BufferSource") {
+      // XXX: For now this is a hack. Consider moving parents' extAttrs into types as the spec says:
+      // https://webidl.spec.whatwg.org/#idl-annotated-types
+      for (const extAttrs of [this.extAttrs, this.parent?.extAttrs]) {
```

Comment:
Could this be `[this.extAttrs, this.parent?.extAttrs].flat()` to avoid the second for loop?

Reply:
yeah, maybe also... to flatten to code a little more:
``` 
for (const extAttr of [this.extAttrs, this.parent?.extAttrs]) {
   if (extAttr.name !== "AllowShared") continue;
   // the rest... 
}
```

Reply:
That way the autofix won't know which array includes the attr and it'll need to rescan 😬